MERN/Topic/Backend/Node/4.1_Import and create modules.md

```diff
@@ -0,0 +1,113 @@
+## What are Node modules ?
+
+In Node. js, Modules are the blocks of encapsulated code that communicates with an external application on the basis of their related functionality
+ Modules can be a single file or a collection of multiples files/folders
+## Why do we use it?
+The major advantage of Modularity is reusability. We don’t need to redevelop the same existing functionality. We can import a Module and reuse that module functionality very easily.
+
+## IMPORTING NODEjs MODULES
+First we install the module using command in the terminal in the your project directory .  
+
+          npm install module_name  
+          
+ ![Screenshot (279)](https://user-images.githubusercontent.com/68144616/134818662-82082f92-9784-44d8-b162-7bf3534caa3b.png)
+ 
+ - After it has been installed you can see  it is mentioned in the dependencies in the package.json
+
+![Screenshot (282)](https://user-images.githubusercontent.com/68144616/134818947-938c759b-8d8a-4252-af01-4f7070f58d81.png)
+
+      
+  
+  
+
+
+ After that,we import it using **require()** function.
+This **require()** call import the specified module and cached into the application so that we don’t need to import it again and again.
+
+### SYNTAX :
+     
+
+          var some-name= require('module-name')     
+      
+         
+
+### EXAMPLE
+•	Import Nodejs built-in “express” module.  
```

Comment:
Correction: express and mongoose are not built-in modules, They both have to be installed before using them.